pkg/etcd/client.go

```diff
@@ -188,7 +188,14 @@ func (c *Client) WatchWithChan(ctx context.Context, outCh chan<- clientv3.WatchR
 		close(outCh)
 		log.Info("WatchWithChan exited")
 	}()
-	var lastRevision int64
+
+	// get initial revision from opts to avoid revision fall back
+	op := &clientv3.Op{}
```

Comment:
Can you briefly explain the algorithm? It would probably be even better if you could extract this piece of code into a unit-testable function.

Reply:
Since this piece of code relies on some implementation details of an external package, we would unit-test it.

Reply:
ok, thanks for your advice~

Reply:
done PTAL~